### ISSUE TITLE
Booleans validate as integers

### ISSUE THREAD

**Issue by mthebridge** [CZ-NIC/yangson#37]
Yangson considers the Boolean `true` and `false` as valid values for integer type nodes.

For example, with this minimal schema:

```
$ cat test.json 
{
    "ietf-yang-library:modules-state": {
      "module-set-id": "",
      "module": [
        {
          "name": "test",
          "revision": "1.0.0",
          "conformance-type": "implement"
        }
      ]
    }
  }
$ cat test.yang 
module test {
    prefix ;
    revision 1.0.0;
    description "Test schema.";
    leaf number {
        type int32;
    }
}
```
Then I'd expect the following code to raise an exception, but it parses happily:

```python
from yangson import DataModel
import yaml

data = DataModel.from_file("./test.json")
yaml_input = yaml.safe_load("'test:number': true")
inst = data.from_raw(yaml_input)
inst.validate()
```

Replacing `true` with any string or non-integer value triggers a `RawTypeError` as expected.

**Comment by mthebridge** [comment]
Suspect https://github.com/CZ-NIC/yangson/blob/acdf40cf56529f1897de94d76ce47a18f10c8edc/yangson/datatype.py#L708 is the culprit - confusingly, Python *does* consider Booleans to be of type Int:
```>>> isinstance(True, int)
True
```

So probably needs some special handling...

**Comment by llhotka** [comment]
Fixed in de4a558895b2caf58a972cb4cfa90b15a2f2be16 -> PyPI package 1.3.48

**Comment by mthebridge** [comment]
Awesome - thanks for the speedy turnaround!
